apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -248,27 +245,53 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
   end
 
   defp retrieve_trigger_configuration(realm_name, trigger_id) do
-    query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(
-        "SELECT value FROM kv_store WHERE group='triggers' AND key=:trigger_id;"
-      )
-      |> DatabaseQuery.put(:trigger_id, trigger_id)
+    statement =
+      "SELECT value FROM #{realm_name}.kv_store WHERE group='triggers' AND key=:trigger_id;"
 
-    with {:ok, client} <- Database.connect(realm: realm_name),
-         {:ok, result} <- DatabaseQuery.call(client, query),
-         [value: trigger_data] <- DatabaseResult.head(result),
-         trigger <- Trigger.decode(trigger_data),
-         {:ok, action} <- Jason.decode(trigger.action) do
+    params = %{"trigger_id" => trigger_id}
+
+    with :ok <- check_realm_name_ok(realm_name),
+         {:ok, prepared} <- Xandra.Cluster.prepare(:xandra, statement),
+         {:ok, result} <- Xandra.Cluster.execute(:xandra, prepared, params),
+         {:ok, action} <- parse_trigger_data(result) do
       {:ok, action}
     else
-      {:error, :database_connection_error} ->
-        Logger.warn("Database connection error.")
+      {:error, error = %Xandra.Error{}} ->
+        Logger.warning("Database error: #{Exception.message(error)}", tag: "database_error")
+
+        {:error, :database_error}
+
+      {:error, error = %Xandra.ConnectionError{}} ->
+        Logger.warning("Database connection error while #{Exception.message(error)}",
+          tag: "database_connection_error"
+        )
+
         {:error, :database_connection_error}
 
       error ->
-        Logger.warn("Error while processing event: #{inspect(error)}")
+        error
+    end
+  end
+
+  defp parse_trigger_data(xandra_page) do
+    with %{"value" => trigger_data} <-
+           Enum.at(xandra_page, 0, {:error, :invalid_query_result, xandra_page}),
```

Comment:
I'm not sure we want to return the whole Xandra page in case of an error here

Reply:
We're not, we're returning `{:error, :trigger_not_found}` from the `else`
Or you mean we don't want to `inspect` it?

Reply:
Right, my mistake